### PR TITLE
Feature/quiz list

### PR DESCRIPTION
**Pull request by wutek**
Quiz list view, with pagination.
Quizes per page are set in variable in controller.

### CODE REVIEW

File: apps/quiz/models.py [hunk 1]
```diff
@@ -34,6 +34,7 @@ class Quiz(db.Model):
 
     description = db.Column(db.String(500))
     questions = db.relationship('Question', backref='quiz', lazy=True)
+    author = db.relationship('User')
```
Comment (mtyton) [inline apps/quiz/models.py:37]: I  see no reason for this field here, if that's an old field (some leftover) it should be removed

File: apps/quiz/views.py [hunk 2]
```diff
@@ -9,15 +9,34 @@
...
 bp = Blueprint('quiz', __name__)
 
 
 class QuizListView(BasePermissionCheckMethodView):
-
+    QUIZZES_PER_PAGE = 2
     template_name = "quiz/quiz_list.html"
 
+    def get_context(self) -> dict:
+        quizzes = Quiz.query.all()
```
Comment (mtyton) [inline apps/quiz/views.py]: gettin all quizzes may not be the best idea, in case there are thousands of quizzes

File: apps/quiz/views.py [hunk 3]
```diff
@@ -9,15 +9,34 @@
...
+        max_page = int(len(quizzes) / self.QUIZZES_PER_PAGE)
+        page = request.args.get('page', default=1, type=int)
+
+        # Validate page number.
+        page = max(1, page)
```
Comment (mtyton) [inline apps/quiz/views.py]: You are basically reinventing the wheel here, how about using this?
[https://flask-sqlalchemy.palletsprojects.com/en/2.x/api/?highlight=paginate#flask_sqlalchemy.BaseQuery.paginate](https://flask-sqlalchemy.palletsprojects.com/en/2.x/api/?highlight=paginate#flask_sqlalchemy.BaseQuery.paginate)
or
[https://blog.miguelgrinberg.com/post/the-flask-mega-tutorial-part-ix-pagination](https://blog.miguelgrinberg.com/post/the-flask-mega-tutorial-part-ix-pagination)

File: apps/quiz/models.py (hunk 1 above)
Comment (wutek) [reply apps/quiz/models.py:37]: There is username under each quiz, thanks to this relationship I can get to it like this `quiz.author.username`. Is there better way?

Comment (mtyton) [reply apps/quiz/models.py:37]: Sure but what you are doing here, is clearly repeating the code, take a look at main branch:
- there already exists the field author_id so there is a relation One to Many, there is  no need to define another relation
- You can achieve this thing you are talking about using @property, simply use author_id to get author instance and return this object in property called author, if

**Review by mtyton — APPROVED** [review]

**Pull request merged**